runhouse/main.py

```diff
@@ -679,6 +311,149 @@ def cluster_list(
+    This only applies to on-demand clusters, and has no effect on self-managed clusters.
+
+    Example:
+        ``$ runhouse cluster up /sashab/rh-basic-cpu``
+
+    """
+    current_cluster = get_cluster_or_local(cluster_name)
+
+    try:
+        current_cluster.up_if_not()
+    except ValueError:
+        console.print("Cluster is not saved in den.")
+    except Exception as e:
+        console.print(f"Failed to bring up the cluster: {e}")
+
+
+@cluster_app.command("down")
+def cluster_down(
+    cluster_name: str = typer.Argument(
+        None,
+        help="The name of cluster to terminate.",
+    ),
+    remove_configs: bool = typer.Option(
+        False,
+        "-rm",
+        "--remove-configs",
+        help="Whether to delete cluster config from Den (default: ``False``).",
+    ),
+):
+    """Terminate cluster if it is not down. No-op if cluster is already down.
+    This only applies to on-demand clusters, and has no effect on self-managed clusters.
+
+    Example:
+        ``$ runhouse cluster down /sashab/rh-basic-cpu``
+    """
+    current_cluster = get_cluster_or_local(cluster_name)
+
+    try:
+        current_cluster.teardown_and_delete() if remove_configs else current_cluster.teardown()
+    except ValueError:
+        console.print("Cluster is not saved in den, could not bring it down.")
+        sky_live_clusters = get_all_sky_clusters()
+        cluster_name = (
+            cluster_name.split("/")[-1] if "/" in cluster_name else cluster_name
+        )
+        if cluster_name in sky_live_clusters:
+            console.print(
+                f"You can bring down the cluster by running [italic bold] sky down {cluster_name} -y"
+            )
+    except Exception as e:
+        console.print(f"Failed to terminate the cluster: {e}")
+
+
+@cluster_app.command("logs")
+def cluster_logs(
+    cluster_name: str = typer.Argument(
+        None,
+        help="Name of cluster to get the logs from. If not specified, we will get the logs of the local cluster.",
+    ),
+    since: Optional[int] = typer.Option(
+        LogsSince.five_min,
+        "--since",
+        help=f"Get the logs since the given timeframe. Available timeframes (minutes): {LogsSince.supported()}",
+    ),
+):
+    """Load the logs of the Runhouse daemon running on a cluster.
+
+    Examples:
+        ``$ runhouse cluster logs /sashab/rh-basic-cpu``
+
+        ``$ runhouse cluster logs /sashab/rh-basic-cpu --since 60``
+
+    """
+    if since and str(since) not in LogsSince.supported().split(", "):
+        console.print(
+            f"[reset]{since} is not a supported since filter, using default ({LogsSince.five_min})."
+        )
+        since = LogsSince.five_min
+
+    current_cluster = get_cluster_or_local(cluster_name)
+
+    if not current_cluster:
+        return
+
+    cluster_uri = rns_client.resource_uri(current_cluster.rns_address)
+
+    presigned_file_resp = requests.get(
```

Comment:
maybe just call it "resp"? user doesn't need to know specifics about it coming from a pre-signed url